openmdao/core/driver.py

```diff
@@ -543,3 +543,46 @@ def run(self, problem):
         system.solve_nonlinear(metadata=metadata)
         for recorder in self.recorders:
             recorder.raw_record(system.params, system.unknowns, system.resids, metadata)
+
+    def generate_docstring(self):
+        """
+        Generates a numpy-style docstring for a user-created Driver class.
+
+        Returns
+        -------
+        docstring : str
+                string that contains a basic numpy docstring.
+        """
+        #start the docstring off
+        docstring = '\t\"\"\"\n'
+
+        if self._params:
+            docstring += '\n\tParams\n\t----------\n'
+
+        if self._params:
+            for key, value in iteritems(self._params):
+                docstring += "    "+key
+                docstring += " : param"
+                docstring += type(value).__name__
+                docstring += "\n" + str(value)
+                docstring += "\n        <Insert description here.>\n"
+
```

Comment:
I wouldn't think that the driver would have any built-in params would it?  The params, constraints, and objectives are all specific to a model, so I wouldn't expect them to be part of the docstring for a driver.